iceoryx_utils/include/iceoryx_utils/posix_wrapper/file_lock.hpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_UTILS_POSIX_WRAPPER_FILE_LOCK_HPP
+#define IOX_UTILS_POSIX_WRAPPER_FILE_LOCK_HPP
+
+#include "iceoryx_utils/cxx/expected.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+#include "iceoryx_utils/design_pattern/creation.hpp"
+
+namespace iox
+{
+namespace posix
+{
+enum class FileLockError
+{
+    INVALID_STATE,
+    NO_FILE_NAME_PROVIDED,
+    LOCKED_BY_OTHER_PROCESS,
+    ACCESS_DENIED,
+    INVALID_FILE_NAME,
+    FILE_EXISTS,
+    QUOTA_EXHAUSTED,
+    INVALID_ARGUMENTS,
```

Comment:
The abstraction should make it impossible to provide invalid arguments to the underlying c calls! Could you please remove this error enum.